src/librustc_typeck/check/wfcheck.rs

```diff
@@ -172,6 +172,18 @@ pub fn check_trait_item(tcx: TyCtxt<'_>, def_id: DefId) {
         _ => None
     };
     check_associated_item(tcx, trait_item.hir_id, trait_item.span, method_sig);
+
+    // Prohibits applying `#[track_caller]` to trait methods
+    for attr in &trait_item.attrs {
```

Comment:
Use `tcx.has_attr(def_id, sym::track_caller)`: https://doc.rust-lang.org/nightly/nightly-rustc/rustc/ty/struct.TyCtxt.html#method.has_attr

Reply:
The code is definitely cleaner, but ISTM that without additional work the error will be reported for the whole trait item, not just the attribute. Should I use `has_attr` and then search for the offending attribute's span within the conditional?

Reply:
Oh right, you want to point at a specific method since that's better UX. Nah, keep it as-is then.